shell-vars.wake

```diff
@@ -20,3 +20,9 @@ global def ifWakeEnv where list =
   match (getenv "WAKE_ENV")
     Some env = if env ==~ where then list else Nil
     None     = list
+
+# Return the list if and only if 'where' equals 'WAKE_ENV'
+global def requireWakeEnv where list =
```

Comment:
I feel like this function is too similar to `ifWakeEnv` and very subtly different, but the names of the functions aren't sufficiently suggestive of the differences between them. Is there a better name we can give this, even if it is grossly verbose? For example, `ifWakeEnvIsDefinedAndEquals`, with the other being defined as `ifWakeEnvIsUndefinedOrEquals`?

Or is this so subtle that we shouldn't actually have a helper and instead just inline exactly what we mean in all the call sites? For example:

```wake
publish myList = match (getenv "WAKE_ENV")
  Some env = if env ==~ "home" then list else Nil
  None = list
```

cc @jackkoenig, @albertchen-sifive, or @hcook if they have ideas on better names here.

Reply:
I do think we should have the helpers regardless of what name they might have

To help with naming, I see the current `requireWakeEnv` as the version that we will eventually use in environment wake rules and that `isWakeEnv` is a transitional function used while we still support wit

Reply:
I guess it's actually the original `isWakeEnv` name that causes the issue, I'd be happy with it having a longer name like `ifWakeEnvWithFallback`, though that would mean some work to change the call sites

Reply:
Maybe we could add the longer name to the code here, but not remove `ifWakeEnv`, and gradually replace things over time? Anyway, no need to dwell on this much further.

Reply:
Actually, one last suggestion, what if we used `whenWakeEnv` instead of `requireWakeEnv`? Ignoring the existence of `ifWakeEnv`, I feel like `requireWakeEnv` almost sounds like it's going to throw an error or return a `Fail` if it doesn't match, but `whenWakeEnv` sounds more like a conditional.

This does nothing to distinguish between the existing `ifWakeEnv`, but maybe will set us up for a better name for the new function with the intent to eventually deprecate the old one's short name?

Reply:
I'm onboard with `whenWakeEnv`